consensus/abci.go

```diff
@@ -143,63 +175,32 @@ func (abci *TendermintABCI) CheckTx(txBytes []byte) tmtAbciTypes.ResponseCheckTx
 
 	from, err := ethTypes.Sender(signer, tx)
 	if err != nil {
-		abci.logger.Error(core.ErrInvalidSender.Error())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrBaseInvalidSignature.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrBaseInvalidSignature.Code), Log: core.ErrInvalidSender.Error()}
+		return common.Address{}, fmt.Errorf("INVALID_SENDER")
 	}
 
 	if tx.Value().Sign() < 0 {
-		abci.logger.Error(core.ErrNegativeValue.Error())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrBaseInvalidInput.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrBaseInvalidInput.Code), Log: core.ErrNegativeValue.Error()}
+		return common.Address{}, fmt.Errorf("INVALID_SIGNATURE")
 	}
 
 	if !abci.checkTxState.Exist(from) {
-		abci.logger.Error(core.ErrInvalidSender.Error())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrBaseUnknownAddress.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrBaseUnknownAddress.Code), Log: core.ErrInvalidSender.Error()}
+		return common.Address{}, fmt.Errorf("UNKNOWN_ADDRESS")
 	}
 
 	if abci.checkTxState.GetNonce(from) != tx.Nonce() {
-		errMsg := fmt.Sprintf("Nonce not strictly increasing. Expected %d got %d", abci.checkTxState.GetNonce(from), tx.Nonce())
-		abci.logger.Error(errMsg)
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrBadNonce.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrBadNonce.Code), Log: errMsg}
+		abci.logger.Error(fmt.Sprintf("Nonce not strictly increasing. Expected %d got %d", abci.checkTxState.GetNonce(from), tx.Nonce()))
+		return common.Address{}, fmt.Errorf("BAD_NONCE")
 	}
 
 	intrinsicGas, err := core.IntrinsicGas(tx.Data(), tx.To() == nil, true)
 	if err != nil {
-		abci.logger.Error(err.Error())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrInternalError.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrInternalError.Code), Log: err.Error()}
+		return common.Address{}, fmt.Errorf("INTRINSIC_GAS_UNKNOWN")
 	}
 
 	if tx.Gas() < intrinsicGas {
-		abci.logger.Error("TX gas is lower than intrinsic gas", "tx_gas", tx.Gas(), "intrinsic_gas", intrinsicGas)
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrInsufficientGas.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrInsufficientGas.Code), Log: err.Error()}
-	}
-
-	currentBalance := abci.checkTxState.GetBalance(from)
-	txCost := tx.Cost()
-	if currentBalance.Cmp(txCost) < 0 {
-		errMsg := fmt.Sprintf("Current balance: %s, TX cost: %s", currentBalance, tx.Cost())
-		abci.logger.Error(errMsg)
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrInsufficientFunds.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrInsufficientFunds.Code), Log: errMsg}
+		return common.Address{}, fmt.Errorf("INTRINSIC_GAS")
```

Comment:
[MINOR] Wrong message. Maybe "NOT_ENOUGH_INTRINSIC_GAS"

Reply:
that's obvious from the name instrinsic but well, I will change it just for you 💋  INSUFFICIENT_INTRINSIC_GAS